testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -433,6 +476,61 @@ pub fn boot_to_epoch_3(
     info!("Bootstrapped to Epoch-3.0 boundary, Epoch2x miner should stop");
 }
 
+fn signer_vote_if_needed(
+    btc_regtest_controller: &BitcoinRegtestController,
+    naka_conf: &Config,
+    signer_sks: &[StacksPrivateKey], // TODO: Is there some way to get this from the TestSigners?
+) {
+    if let Some(mut signers) = naka_conf.self_signing() {
+        // When we reach the next prepare phase, submit new voting transactions
+        let block_height = btc_regtest_controller.get_headers_height();
+        let reward_cycle = btc_regtest_controller
+            .get_burnchain()
+            .block_height_to_reward_cycle(block_height)
+            .unwrap();
+        let prepare_phase_start = btc_regtest_controller
+            .get_burnchain()
+            .pox_constants
+            .prepare_phase_start(
+                btc_regtest_controller.get_burnchain().first_block_height,
+                reward_cycle,
+            );
+
+        // TODO: Check if the vote has already happened
```

Comment:
I think you should do this before merging. It's straightforward to do via the read-only-call API.

Reply:
Added in 91a1fd95e.